example/55-managedseed-gardenlet.yaml

```diff
@@ -34,6 +34,17 @@ spec:
 #     additionalVolumeMounts: []
 #     env: []
 #     vpa: true
+#     imageVectorOverwrite:
+#     - name: pause-container
+#       repository: my-custom-image-registry/pause-amd64
+#       tag: "3.1"
+#       targetVersion: ">= 1.12"
+#     componentImageVectorOverwrites:
+#     - name: etcd-druid
+#       imageVector:
+#       - name: etcd
+#         repository: my-custom-image-registry/etcd
+#         tag: latest
```

Comment:
See https://github.com/gardener/gardener/pull/3627#discussion_r584520850 for the motivation. Even though this is not expected to be used (much) in managed seeds (and this is why it wasn't implemented originally), we want to reuse the `GardenletDeployment` type (and its defaults, validation, etc.) for the gardenlet landscaper component, and there it's a must. So after discussing this with @timebertt and @danielfoehrKn , we decided to add the 2 fields to this shared type. I think it doesn't hurt for managed seeds as well, and it even makes the current "merge with parent" implementation more straightforward and consistent.

Regarding "blowing-up" resources - I see your point, but referencing `ConfigMap` would contradict the entire idea of having strongly typed image vectors in the first place. Instead, if separate resources are to be used, they should rather be dedicated image vector resources, e.g. `ImageVector` and `ComponentImageVectors`. Perhaps this makes sense, we should discuss this idea further.

Reply:
> Regarding "blowing-up" resources - I see your point, but referencing ConfigMap would contradict the entire idea of having strongly typed image vectors in the first place. Instead, if separate resources are to be used, they should rather be dedicated image vector resources, e.g. ImageVector and ComponentImageVectors. Perhaps this makes sense, we should discuss this idea further.

I don't know, it feels odd to introduce such custom resources. This is clearly configuration data for the gardenlet, so having a `ConfigMap` seems way more appropriate.
I can imagine that inlining this in the `Imports` resource for the gardenlet landscaper component might also become a problem for the Landscaper and its deployers in the long run.
 
What was again the motivation for those strongly typed fields? Was it just to have early validation? If yes, why isn't it "good enough" if the gardenlet fails after start-up if its configuration is broken? This is what is happening for most components, including the kubelet. Yes, we could have argued similarly for the inlined `GardenletConfiguration` in `ManagedSeed`, but it's still different because it grow arbitrarily in size like the image vectors overrides.

Reply:
Ok, agree with the point about avoiding too large resources because of an inlined ImageVector.

> What was again the motivation for those strongly typed fields? Was it just to have early validation?

The main motivation for having strongly typed fields was to have a clean API for values of which we already know the concrete structure, i.e. not representing image vectors as strings, because there is a well-known structure for them.
This would in turn then allow early validation of the structural integrity of the image vecotr, which is desirable FMPOV.

> If yes, why isn't it "good enough" if the gardenlet fails after start-up if its configuration is broken? This is what is happening for most components, including the kubelet.

There is also other configuration data we try to validate as early on as possible, e.g. Shoot audit policy config maps. We do this, because it will take quite some time until the error surfaces if we don't fail early on. Same applies to the image vector IMO. If we create a new shoot and deploy a gardenlet into it after ~8m and fail only because there is a space missing in the YAML, it would be quite the bad UX.
Another scenario where it would be even more problematic is when rolling existing gardenlets of a `ManagedSeedSet`. If there is some YAML-format issue and new gardenlet instances won't be able to start because of that, the existing Seeds won't have any functional gardenlet, although the error could have been caught very early on.